pyocd/probe/pydapaccess/interface/pyusb_backend.py

```diff
@@ -343,7 +351,7 @@ def __call__(self, dev):
 
             # Now read the product name string.
             device_string = dev.product
-            if (device_string is None) or ("CMSIS-DAP" not in device_string):
+            if (device_string is None) or not any(map(lambda s: s in device_string, ACCEPTED_DEVICE_NAMES)):
```

Comment:
Quick Python note: `any()` is the right idea! However, you can simply the inner expression using a comprehension as `any(s in device_string for s in ACCEPTED_DEVICE_NAMES)`. It's faster, too. 😄